api4/websocket.go

```diff
@@ -5,9 +5,13 @@ package api4
 
 import (
 	"net/http"
+	"time"
 
-	"github.com/gorilla/websocket"
+	"github.com/mattermost/mattermost-server/v5/mlog"
 	"github.com/mattermost/mattermost-server/v5/model"
+
+	"github.com/gobwas/ws"
+	"github.com/mailru/easygo/netpoll"
```

Comment:
Given this is a change touching a very delicate part of our backend I have to ask:

What's your general opinion on these dependencies? Are they used in production elsewhere? Are they considered stable? Should we do a quick review before merging or do you think it's enough to test it internally (plus possibly a few long running load-tests)?

Reply:
Yes, I think we should do a review of the package. It's essentially an epoll API. The code is fairly self-contained here: https://github.com/mailru/easygo/blob/3c14a0dc985f4cd7899588176a176809055d2b3b/netpoll/epoll.go.

I don't have hands-on experience with implementing event loops myself. So it would be good to have multiple pair of eyes go through it :)

The only catch that I just realized now is that we would be losing Windows support. I don't know how strongly we supported native Windows. But since it doesn't have kqueue/epoll, this will straightaway throw an error.

Reply:
I took a look. Looks simple enough where you just add/remove fds to an epoll instance. The core event loop logic is here: https://github.com/mailru/easygo/blob/3c14a0dc985f4cd7899588176a176809055d2b3b/netpoll/epoll.go#L245-L279. Let me know if you see any issues with it.

Reply:
Alright, getting back to Windows support (yeah), should we include a PM in the decision?

Reply:
Yeah, let me talk to Katie. I do see windows builds being part of https://latest.mattermost.com/. So there might be some pushback :confounded: Let's see.